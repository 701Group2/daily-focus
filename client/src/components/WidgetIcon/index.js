import React, { useState } from "react";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Grid from "@material-ui/core/Grid";

import styles from "./style.module.scss";

function WidgetIcon({ image, label, selectedWidgets, setSelectedWidgets }) {
    const [selected, setSelected] = useState(false);

    const handleClick = () => {
        setSelected(!selected);

        // if (!selectedWidgets.includes(label)) {
        //     //checking weather array contain the id
        //     setSelectedWidgets([...selectedWidgets, label]);
        // } else {
        //     setSelectedWidgets(
        //         selectedWidgets.filter((selectedWidget) => selectedWidget !== label)
        //     ); //deleting
        // }
    };

    return (
        <Grid item xs={6}>
            <Card
                className={`${styles.card} ${selected && styles.selectedCard}`}
                onClick={handleClick}
                raised
            >
                <img src={image} className={styles.media} alt={label} />
                <CardContent className={styles.cardContent}>{label}</CardContent>
            </Card>
        </Grid>
    );
}

export default WidgetIcon;
