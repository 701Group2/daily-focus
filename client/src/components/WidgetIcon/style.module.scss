.card {
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: 350px;
}

.cardContent {
    text-align: center;
}

.media{
    display: block;
    max-width:250px;
    max-height:250px;
    width: auto;
    height: auto;
}

.selectedCard{
    box-sizing: border-box;
    border: 3px solid red;
    border-radius: 10px;
}