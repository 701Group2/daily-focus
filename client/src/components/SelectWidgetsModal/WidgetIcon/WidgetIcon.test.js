import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";

import WidgetIcon from "./";

test("it renders correctly", () => {
    const shallowRenderer = new ShallowRenderer();
    const snapshotComponent = shallowRenderer.render(<WidgetIcon />);
    expect(snapshotComponent).toMatchSnapshot();
});
