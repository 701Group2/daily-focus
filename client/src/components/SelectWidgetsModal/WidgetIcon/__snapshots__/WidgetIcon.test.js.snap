// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  item={true}
  xs={6}
>
  <WithStyles(ForwardRef(Card))
    className="card false"
    onClick={[Function]}
    raised={true}
  >
    <img
      className="media"
    />
    <WithStyles(ForwardRef(CardContent))
      className="cardContent"
    />
  </WithStyles(ForwardRef(Card))>
</WithStyles(ForwardRef(Grid))>
`;
